src/runtime_src/core/pcie/linux/device_linux.h

```diff
@@ -86,9 +93,16 @@ class device_linux : public shim<device_pcie>
   {
     return xrt::shim_int::alloc_bo(get_device_handle(), userptr, size, xcl_bo_flags{flags}.flags);
   }
-  ////////////////////////////////////////////////////////////////
+
+protected:
+  std::shared_ptr<pci::dev>
+  get_dev() const
+  {
+    return m_pcidev;
+  }
```

Comment:
Suggest to return `pci::dev*` and not the shared pointer
```suggestion
  pci::dev*
  get_dev() const
  {
    return m_pcidev.get();
  }
```
The shared_ptr is for sharing ownership, not of the class I see to this function needs to share ownership of the pcidev.

Reply:
fixed.